### PR TITLE
Resolving required dependency during generate command

### PR DESCRIPTION
**Pull request by shivankurchavan**
## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by shivankurchavan** [comment]
@techsavvyash please review

**Comment by Savio629** [comment]
fixes: https://github.com/SamagraX-Stencil/stencil-cli/issues/8

**Comment by shivankurchavan** [comment]
@techsavvyash any reviews?

File: src/lib/service-user/service-user.factory.ts [hunk 1]
```diff
@@ -48,6 +50,19 @@ function addImportToModule(options: ServiceOptions): Rule {
 
     tree.overwrite(options.module, content);
 
+    const nodeDependencyRef = getPackageJsonDependency(
+      tree,
+      '@techsavvyash/user-service',
```
Comment (techsavvyash) [inline src/lib/service-user/service-user.factory.ts]: Check [`packages/user-service`](https://github.com/SamagraX-stencil/stencil/tree/main/packages/user-service/package.json) -- this should be mentioning the `package-name` as `@samagra-x/user-service` and hence we need to use the same.

File: src/lib/resolver/resolver.factory.ts [hunk 2]
```diff
@@ -97,6 +100,20 @@ function addDeclarationToModule(options: ProviderOptions): Rule {
       options.module,
       declarator.declare(content, options as DeclarationOptions),
     );
+
+    const DependencyRef = getPackageJsonDependency(
+      tree,
+      '@nestjs/graphql',
```
Comment (techsavvyash) [inline src/lib/resolver/resolver.factory.ts:106]: Where do we require graphql?

**Review by techsavvyash — CHANGES_REQUESTED** [review]

File: src/lib/service-user/service-user.factory.ts (hunk 1 above)
Comment (shivankurchavan) [reply src/lib/service-user/service-user.factory.ts]: Noted

File: src/lib/resolver/resolver.factory.ts (hunk 2 above)
Comment (shivankurchavan) [reply src/lib/resolver/resolver.factory.ts:106]: Idk, it was mentioned already so made the provisions for imports.

**Comment by techsavvyash** [comment]
@shivankurchavan if this PR is no longer required then please close this off.

**Pull request merged**